operator/src/main/java/oracle/kubernetes/operator/helpers/PodStepContext.java

```diff
@@ -359,6 +361,15 @@ protected Step patchPod(V1Pod currentPod, Step next) {
     KubernetesUtils.addPatches(
         patchBuilder, "/metadata/annotations/", getAnnotations(currentPod), getPodAnnotations());
 
+    if (introspectVersionChanged(currentPod)) {
```

Comment:
This is still much more complex than is needed. Look at `PodStepContext.createMetadata()`. This is where we define the labels and annotations for server pods. To add a new label, all you need to do is define it there. The rest of the code will automatically handle detection of differences between what is supposed to be and what we find.

Reply:
The label was added in the createPodModel method but in the non hashed part. If you look into getPodLabels you'll see that the labels that are to be patched here are the custom defined labels in the server specs.